araig_calculators/src/loggers/rosbagger_class.py

```diff
@@ -3,84 +3,73 @@
 import rospy
 import subprocess, shlex
 from araig_msgs.msg import BoolStamped
+from base_classes.base_logger import BaseLogger
 import threading
 
-class RosbaggerClass(object):
-    def __init__(self,  start_topic = "/start",
-                        stop_topic = "/stop",
-                        stop_offset = 2,
-                        filename ="test"):
-        self.filename_ = filename 
-        self._stop_offset = stop_offset
-        self.start_mutex_ = threading.Lock()
-        self.stop_mutex_ = threading.Lock()
-        self._start_recording = False
-        self._stop_recording_ = False
-        self._rate = rospy.Rate(10)
+class RosbaggerClass(BaseLogger):
+    def __init__(self):
 
-        rospy.Subscriber(start_topic, BoolStamped, self.start_callback)
-        rospy.Subscriber(stop_topic, BoolStamped, self.stop_callback)
+        self.node_name = rospy.get_name()
+        param_list = [
+            self.node_name + "/start_offset",
```

Comment:
This is a potential concern area regarding namespaces. If this node is launched from another namespace, would these params still be found? The best way to read params is through private node handles, but I do not know how this is done in python.

Reply:
In fact I would say that since you have made a nice base class for all loggers, we should have there a nifty function that goes through the list of params for all loggers and ensures that they exist and can be read. This function needs to be called by constructor before doing anything else.

In double fact, this function needs to be implemented for every base class that reads rosparams. This will take some time for sure, but it will save us a LOT of headache going forward, as we will never need to worry about params and namespace.

Reply:
https://github.com/ipa320/araig_test_stack/issues/28